magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/PolicyAcquisitionServiceImpl.java

```diff
@@ -214,8 +211,28 @@ private String getProjectNameFromRepository(String repository) {
   }
 
   private Path getTargetProjectDirectoryPath(String repository) {
-    return Path.of(policyConfig.getRoot().replace("~", System.getProperty("user.home"))
-      + "/"
-      + getProjectNameFromRepository(repository));
+    if (isGitRepository(repository)) {
+      String[] tokens = repository
+        .replace("git@", "")
+        .replace("https://", "")
+        .replace(":", "/")
+        .split("/");
+      return
+        Path.of(String.format(
+            "%s/%s/%s/%s",
+            policyConfig.getRoot().replace("~", System.getProperty("user.home")),
+            tokens[0],
+            tokens[1],
+            getProjectNameFromRepository(repository)));
+    } else {
+      return Path.of(String.format(
+        "%s/%s",
+        policyConfig.getRoot().replace("~", System.getProperty("user.home")),
+        getProjectNameFromRepository(repository)));
+    }
+  }
+
+  private boolean isGitRepository(String repository) {
+    return repository.startsWith("git@") || repository.startsWith("https://");
```

Comment:
`startsWith("https://")` -> `startsWith("https://github")` ?

This way any link starting with `https` will be accepted as git repo and tried to be pulled